docs/docs/tutorials/basics/part1.md

```diff
@@ -0,0 +1,200 @@
+see how our SQL program can be instantiated with any of these data sources, or
+even multiple data sources connected to the same table.
+
+## Step 2. Write queries
+
+We would like to compute the lowest price for each part
+across all vendors.  Add the following statements to your SQL program
+
+```sql
+-- Lowest available price for each part across all vendors.
+create view LOW_PRICE (
+    part,
+    price
+) as
+    select part, MIN(price) as price from PRICE group by part;
+
+-- Lowest available price for each part along with part and vendor details.
+create view PREFERRED_VENDOR (
+    part_id,
+    part_name,
+    vendor_id,
+    vendor_name,
+    price
+) as
+    select
+        PART.id as part_id,
+        PART.name as part_name,
+        VENDOR.id as vendor_id,
+        VENDOR.name as vendor_name,
+        PRICE.price
+    from
+        PRICE,
+        PART,
+        VENDOR,
+        LOW_PRICE
+    where
+        PRICE.price = LOW_PRICE.price AND
+        PRICE.part = LOW_PRICE.part AND
+        PART.id = PRICE.part AND
+        VENDOR.id = PRICE.vendor;
+```
+
+In Feldera we write queries as SQL views.  Views can be defined in terms of
+tables and other views, making it possible to express deeply nested queries.  In
+this example, the `PREFERRED_VENDOR` view is expressed in terms of the
+`LOW_PRICE` view.
+
+## Step 3. Run the program
+
+In order to run our SQL program, we must instantiate it as part of a _pipeline_.
+Navigate to the `Pipelines` section and click `ADD PIPELINE`.  Give the new
+pipeline a name, e.g., "Supply Chain Test Pipeline" and select "Supply Chain
+Analytics" from the list of SQL programs.
+
+The selected program is visualized as a rectangle with a little blue stub for each
+table and view declared in the program.  These can be used to connect data sources
+and sinks.  For the time being, we will run our pipeline without any sources or
+sinks.  We will build more exciting pipelines in the next part of the tutorial.
+
+![Pipeline Builder](pipeline-builder.png)
+
+Go back to the pipelines view (click on `Pipelines` in the navigation bar
+on the left).  Your newly created pipeline should appear in the list.  Click the
+play icon <icon icon="bx:play-circle" /> next to the pipeline.
+
+The pipeline is now running and is ready to process inputs; however since we
+have not connected any data sources to it, no data has been received yet.  Let
+us add some manually.
+
+## Step 4. Populate tables manually
+
+Expand the runtime state of the pipeline by clicking the chevron icon <icon
+icon="bx:chevron-down" /> on the left.  You should see the list of tables and
+views defined in your program.  Click on the <icon icon="bx:upload" /> icon next
+to the `PART` table.  This will open the view where you can insert new rows to
+the table using a configurable random data generator (feel free to play around
+with it!) or by entering the data manually.  For example, I created the
+following rows:
```

Comment:
Jarring change of point of view here from "we" and "you" to "I". Maybe:
```suggestion
with it!) or by entering the data manually.  For example, you might add the
following rows:
```

Reply:
Also, I found the interface confusing here. When I first found got to it, I thought I should click on "Insert Rows" to insert some rows, but it didn't do anything. After clicking around, I figured out that to insert data manually, I had to click on "Add Default Row" or "+1 Random Row" or "+5 Random Rows" and then edit whatever showed up and *then* click on "Insert Rows". I think the UI can be improved, but for now maybe we should explain it.

Reply:
Is the problem just how the buttons are named? That can be fixed easily just let me know. 
If you want to change the UI flow maybe open an issue we can discuss a better approach.

Reply:
I agree, this can be a bit confusing.  No sure what's the best way to make it clear that you need to press `Insert Rows` to actually send the data to the table.

Reply:
Not convinced `Confirm Insertions` is better. How about `Insert in ${tablename}` and disable it when empty?

Reply:
Sure.